cars-assemble: Rework to cover only numbers & arithmetic-operators

kahgoh opened the pull request:
Closes #1936

The `conditionals-if` and `comparison` concepts were removed from this
exercise because they are now covered by the vehicle-purhcase exercise
(added by ae2f5e7).

andrerfcsantos commented on exercises/concept/cars-assemble/.meta/design.md:
```diff
@@ -6,17 +6,14 @@ The goal of this exercise is to teach the student how to deal with numbers in Go
 ## Learning objectives
 
-- This should introduce the `int` and `float64` types (be basic number types in Go).
+- This should introduce the `int` and `float64` types (the basic number types in Go).
```
The exercises teaches about `uint` too. It should be added here or alternatively, we should make this exercise not teach `uint`.

kahgoh replied:
I've updated `design.md` to add `uint`.

andrerfcsantos commented on exercises/concept/cars-assemble/cars_assemble_test.go:
```diff
@@ -7,46 +7,53 @@ import (
 		t.Run(tt.name, func(t *testing.T) {
-			got := SuccessRate(tt.speed)
+			got := CalculateWorkingCarsPerHour(tt.productionRate, tt.successRate)
 			if math.Abs(got-tt.want) > FloatEqualityThreshold {
```
To ensure some consistency on the track, implement this float comparison like the [interest is interesting](https://github.com/exercism/go/blob/main/exercises/concept/interest-is-interesting/interest_is_interesting_test.go#L10) exercise.

kahgoh replied:
I've updated the test to use the same pattern.

andrerfcsantos commented on exercises/concept/cars-assemble/.meta/exemplar.go:
```diff
@@ -1,44 +1,23 @@
 }
 
-// CalculateProductionRatePerMinute describes how many working items are
+// CalculateWorkingCarsPerMinute describes how many working items are
```
```suggestion
// CalculateWorkingCarsPerMinute describes how many working cars are
```

kahgoh replied:
Applied as is

andrerfcsantos commented on exercises/concept/cars-assemble/cars_assemble.go:
```diff
@@ -1,25 +1,19 @@
-}
-
-// CalculateProductionRatePerMinute describes how many working items are
+// CalculateWorkingCarsPerMinute describes how many working items are
```
```suggestion
// CalculateWorkingCarsPerMinute describes how many working cars are
```

kahgoh replied:
Applied as is.

andrerfcsantos reviewed (changes requested):
Thanks for taking on this task! Left some suggestions.

andrerfcsantos reviewed (changes requested):
Sorry it took me a while to review this. Although the suggested changes are marked as resolved, it seems they were not applied. Did something happen or did you choose to not accept them?

kahgoh commented:
> Sorry it took me a while to review this.

That's okay ;)

> Although the suggested changes are marked as resolved, it seems they were not applied. Did something happen or did you choose to not accept them?

That's weird ... I just had a look (in the _Files changed_ tab) and as I far as I can tell, I applied the changes and I can see changes reflected. Although ... I didn't applied all the changes exactly as originally suggested - I changed some of the wording, punctuation and formatting (some via later commits). But they should still be there in spirit. I'll go back and add comments on the ones I've changed to indicate which ones I changed.

Was there a particular change you haven't been able to find?

kahgoh commented on exercises/concept/cars-assemble/.docs/hints.md:
```diff
@@ -4,24 +4,23 @@
 
 - [Basic types][basic types] and [Type conversions][type conversions] tutorials.
```
I've applied the change with the wording "To learn more" instead of "To know more".

kahgoh commented on exercises/concept/cars-assemble/.docs/hints.md:
```diff
@@ -4,24 +4,23 @@
 
-- Use if conditionals to return the values in the table.
+- To calculate, divide the percentage by 100 and multiply the result by the number of cars produced.
+- When multiple two numbers together, they both need to be of the same type.
```
I applied the change with the addition of parenthesis around "passed as an argument", i.e.:

> The percentage (passed as an argument) is a number between 0-100

kahgoh commented on exercises/concept/cars-assemble/.docs/hints.md:
```diff
@@ -4,24 +4,23 @@
+## 2. Calculate the number of working cars produced per minute
 
 - Use the `CalculateProductionRatePerHour` function.
+- Remember, there are 60 minutes in an hour.
```
Applied with the addition of a full stop on the second line, i.e:

> Knowing the production per hour of cars, you can get the production per minute by dividing the production per hour by 60 (the number of minutes in an hour)**.**

kahgoh commented on exercises/concept/cars-assemble/.docs/instructions.md:
```diff
@@ -1,67 +1,51 @@
-In principle, the production increases linearly.
-So with the speed set to `4`, it should produce `4 * 221 = 884` cars per hour.
-However, higher speeds increase the likelihood that faulty cars are produced, which then have to be discarded.
+## 1. Calculate the number of working cars produced per hour
```
Applied as is, without any further changes

kahgoh commented on exercises/concept/cars-assemble/.docs/instructions.md:
```diff
@@ -1,67 +1,51 @@
-However, higher speeds increase the likelihood that faulty cars are produced, which then have to be discarded.
+## 1. Calculate the number of working cars produced per hour
+The assembly line's speed is can be changed.
+But, a higher production rate can lead to more errors and, therefore, a lower success rate.
```
Applied, but with each sentence on a separate line. I was under the impression this is how it is mostly formatted elsewhere?

kahgoh commented on exercises/concept/cars-assemble/.docs/instructions.md:
```diff
@@ -1,67 +1,51 @@
+So the cost for 37 cars is:
+3*95,000+2*29,000+10,000=353,000
+
+Implement the function `CalculateCost` that calculates the cost of producing a number of cars, regardless of whether they are successful:
```
Yup, I've simplified the exercise and removed the groups of 3.

junedev commented:
@kahgoh Sorry for the wait. @andrerfcsantos I very busy at the moment. He will get back to you once he has some more time.

kahgoh commented:
@junedev No worries, thank you for letting me know ;)

andrerfcsantos reviewed (changes requested):
I reviewed the previous changes and they seem to be OK. It must be formatting changes that threw me off originally and made the diff look weird.

I left a few more final suggestions.

Thanks again for your patience.

kahgoh commented:
The latest suggestions look good. I've just applied them.

andrerfcsantos reviewed (approved):
🎉🚀
